crates/typst/src/foundations/repr.rs

```diff
@@ -74,17 +74,34 @@ pub fn format_int_with_base(mut n: i64, base: i64) -> EcoString {
 
 /// Converts a float to a string representation with a specific precision and a
 /// suffix, all with a single allocation.
-pub fn format_float(mut value: f64, precision: Option<u8>, suffix: &str) -> EcoString {
+pub fn format_float(
+    mut value: f64,
+    precision: Option<u8>,
+    force_separator: bool,
+    suffix: &str,
+) -> EcoString {
     if let Some(p) = precision {
         let offset = 10_f64.powi(p as i32);
         value = (value * offset).round() / offset;
     }
+    let float_suffix = if force_separator && value % 1.0 == 0.0 { ".0" } else { "" };
```

Comment:
`value.fract() == 0.0` seems a bit simpler. but with both approaches I'm not 100% sure whether it will always produce the desired result or whether the std formatting that strips the .0 does something else

Reply:
I've used `Display` and `Debug` for `f64` instead: `Debug` always uses a decimal separator, while `Display` only does if necessary.